rust/xaynet-mobile/src/participant.rs

```diff
@@ -293,4 +310,21 @@ impl Participant {
         });
         self.should_set_model = false;
     }
+
+    /// Retrieve the current global model, if available.
+    pub fn global_model(&mut self) -> Option<Model> {
```

Comment:
Here I'm also not sure if we should return `Result<Option<Model>>` or just `Option<Model>`

Reply:
i changed it to `Result<Option<Model>>` so that the caller can retry if the call failed due to a IO error